app/ts-cli/geminicli/export.go

```diff
@@ -0,0 +1,1786 @@
+package geminicli
+
+import (
+	"bytes"
+	"compress/gzip"
+	"context"
+	"crypto/tls"
+	"encoding/binary"
+	"encoding/json"
+	"fmt"
+	"github.com/VictoriaMetrics/VictoriaMetrics/lib/encoding"
+	"github.com/golang/snappy"
+	"github.com/openGemini/openGemini/engine"
+	"github.com/openGemini/openGemini/engine/immutable"
+	"github.com/openGemini/openGemini/engine/index/tsi"
+	"github.com/openGemini/openGemini/lib/bufferpool"
+	"github.com/openGemini/openGemini/lib/config"
+	"github.com/openGemini/openGemini/lib/errno"
+	"github.com/openGemini/openGemini/lib/fileops"
+	"github.com/openGemini/openGemini/lib/index"
+	"github.com/openGemini/openGemini/lib/record"
+	"github.com/openGemini/openGemini/lib/util"
+	"github.com/openGemini/openGemini/lib/util/lifted/vm/protoparser/influx"
+	"github.com/openGemini/opengemini-client-go/opengemini"
+	"github.com/vbauerster/mpb/v7"
+	"github.com/vbauerster/mpb/v7/decor"
+	"io"
+	"io/fs"
+	"log"
+	"math"
+	"net"
+	"os"
+	"path"
+	"path/filepath"
+	"sort"
+	"strconv"
+	"strings"
+	"time"
+)
+
+const (
+	tsspFileExtension    = "tssp"
+	walFileExtension     = "wal"
+	csvFormatExporter    = "csv"
+	txtFormatExporter    = "txt"
+	remoteFormatExporter = "remote"
+	resumeFilePrefix     = "resume_"
+	dirNameSeparator     = "_"
+)
+
+var (
+	mpbProgress         = mpb.New(mpb.WithWidth(100))
+	ResumeJsonPath      string
+	ProgressedFilesPath string
+)
+
+// CreateNewProgressFolder init ResumeJsonPath and ProgressedFilesPath
+func CreateNewProgressFolder() error {
+	home, err := os.UserHomeDir()
+	if err != nil {
+		return err
+	}
+	targetPath := filepath.Join(home, ".ts-cli", time.Now().Format("2006-01-02_15-04-05.000000000"))
+	err = os.MkdirAll(targetPath, os.ModePerm)
+	if err != nil {
+		return err
+	}
+	// create progress.json
+	progressJson := filepath.Join(targetPath, "progress.json")
+	ResumeJsonPath = progressJson
+	// create progressedFiles
+	progressedFiles := filepath.Join(targetPath, "progressedFiles")
+	ProgressedFilesPath = progressedFiles
+	return nil
+}
+
+// ReadLatestProgressFile reads and processes the latest folder
+func ReadLatestProgressFile() error {
+	home, err := os.UserHomeDir()
+	if err != nil {
+		return err
+	}
+	baseDir := filepath.Join(home, ".ts-cli")
+	var dirs []string
+	err = filepath.Walk(baseDir, func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+		if !info.IsDir() || path == baseDir {
+			return nil
+		}
+		dirs = append(dirs, path)
+		return nil
+	})
+	if err != nil {
+		return err
+	}
+	sort.Slice(dirs, func(i, j int) bool {
```

Comment:
`Sort.Strings` may be more efficient if you only want to take the first or last element of a sorted string slice.